pkg/testworkflows/testworkflowcontroller/watchers/commons.go

```diff
@@ -0,0 +1,416 @@
+package watchers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"github.com/kubeshop/testkube/cmd/testworkflow-init/data"
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	"github.com/kubeshop/testkube/pkg/testworkflows/testworkflowprocessor/action/actiontypes"
+	"github.com/kubeshop/testkube/pkg/testworkflows/testworkflowprocessor/action/actiontypes/lite"
+)
+
+const (
+	defaultListTimeoutSeconds  = int64(240)
+	defaultWatchTimeoutSeconds = int64(365 * 24 * 3600)
+	InitStepRef                = "tktw-init"
```

Comment:
It is in 3 different places - `testworkflows/testworkflowcontroller`, `testworkflows/testworkflowcontroller/watchers` and `api/v1/testkube` - I'm not sure where would be appropriate place. For sure either `api/v1/testkube` needs it separate, or it should be completely outside of these modules (to avoid circular dependency)